vkontakte/content/contents/code/vkontakte.js

```diff
@@ -1,398 +1,339 @@
+            "widget": Tomahawk.readBase64( "config.ui" ),
+            fields: [{
+                name: "email",
+                widget: "email_edit",
+                property: "text"
+            }, {
+                name: "password",
+                widget: "password_edit",
+                property: "text"
+            }]
+        };
+    },
 
-            // User logged off from vkontakte
-            if(xhr.responseText.match(/progress\.gif/)){
-                delete VK.search_method
+    newConfigSaved: function() {
+        var config = this.getUserConfig();
 
-                VK.search(artist, song, duration, callback)
+        var changed = 
+            this._email !== config.email ||
+            this._password !== config.password;
 
-                return
-            }
+        if (changed) {
+            this.init();
+        }
+    },
 
-            var container = document.getElementById('vk_search');
-            response = xhr.responseText.replace(/^<!--/, '').replace(/->->/g, '-->');
+    testConfig: function (config) {
+        return this._getLoginPromise(config).catch(function (error) {
+            throw new Error('Invalid credentials');
+        });
+    },
 
-            container.innerHTML = response.split('<!>').join('');
+    init: function() {
+        var config = this.getUserConfig();
 
-            var audio_data = []
-            var audio_rows = container.querySelectorAll('div.audio') 
+        this._email = config.email;
+        this._password = config.password;
 
-            for(var i=0; i<audio_rows.length; i++){
-                if(i>10) break
+        if (!this._email || !this._password) {
+            Tomahawk.reportCapabilities(TomahawkResolverCapability.NullCapability);
+            //This is being called even for disabled ones
+            //throw new Error( "Invalid configuration." );
+            Tomahawk.log("Invalid Configuration");
+            return;
+        }
 
-                var url_data = audio_rows[i].querySelector('input[type=hidden]').value.split(',');
-                var url = url_data[0];
-                var duration = url_data[1];
+        return this._login(config);
+    },
 
-                var title = audio_rows[i].querySelector('td.info span');
+    _apiCall: function (api, params) {
+        if (!this.logged_in) {
+            return this._defer(this._apiCall, [api, params], this);
+        } else if (this.logged_in === 2) {
+            throw new Error('Failed login, cannot _apiCall.');
+        }
 
-                audio_data.push({
-                  artist: audio_rows[i].querySelector('td.info a').innerHTML,
-                  title: title.textContent.trim(),
-                  duration: duration,
-                  url: url
-                })
+        params['access_token'] = this._access_token;
+        params['v']            = this.API_VERSION;
+
+        return Tomahawk.get("https://api.vk.com/method/" + api, {
+            data: params
+        }).then(
```

Comment:
Could you elaborate on those style comments please?
I haven't seen any code style guidelines for resolvers in the repo so it is not clear to me which ones I violated. Thanks.